constraintlayout/constraintlayout/src/main/java/androidx/constraintlayout/widget/ConstraintLayoutStatistics.java

```diff
@@ -0,0 +1,264 @@
+        title = title.substring(title.length() - MAX_WORD);
+        title += " = ";
+        return "CL Perf: " + title + value;
+    }
+
+    /**
+     * log a summary of the stats compared to another statics
+     * @param tag used in Log.v(tag, ...)
+     * @param prev the previous stats to compare to
+     */
+    @SuppressLint("LogConditional")
+    public void logSummary(String tag, ConstraintLayoutStatistics prev) {
+        if (prev == null) {
+            log(tag);
+            return;
+        }
+        DecimalFormat df = new DecimalFormat("###.000");
+        StackTraceElement s = new Throwable().getStackTrace()[1];
+
+        Log.v(tag, "CL Perf: -=  Performance .(" + s.getFileName() + ":" + s.getLineNumber() + ")  =- ");
+        Log.v(tag, compare(df, prev, DURATION_OF_CHILD_MEASURES));
+        Log.v(tag, compare(df, prev, DURATION_OF_LAYOUT));
+        Log.v(tag, compare(df, prev, DURATION_OF_MEASURES));
+        Log.v(tag, compare(prev, NUMBER_OF_LAYOUTS));
+        Log.v(tag, compare(prev, NUMBER_OF_ON_MEASURES));
+        Log.v(tag, compare(prev, NUMBER_OF_CHILD_VIEWS));
+        Log.v(tag, compare(prev, NUMBER_OF_CHILD_MEASURES));
+        Log.v(tag, compare(prev, NUMBER_OF_VARIABLES));
+        Log.v(tag, compare(prev, NUMBER_OF_EQUATIONS));
+        Log.v(tag, compare(prev, NUMBER_OF_SIMPLE_EQUATIONS));
+    }
+
+    /**
+     * get the value of a statistic
+     * @param type
+     * @return
+     */
+    public long getValue(int type) {
+        switch (type) {
+            case NUMBER_OF_LAYOUTS:
+                return mMetrics.mNumberOfLayouts;
+            case NUMBER_OF_ON_MEASURES:
+                return mMetrics.mMeasureCalls;
+            case NUMBER_OF_CHILD_VIEWS:
+                return mMetrics.mChildCount;
+            case NUMBER_OF_CHILD_MEASURES:
+                return mMetrics.mNumberOfMeasures;
+            case DURATION_OF_CHILD_MEASURES:
+                return mMetrics.measuresWidgetsDuration ;
+            case DURATION_OF_MEASURES:
+                return mMetrics.mMeasureDuration;
+            case DURATION_OF_LAYOUT:
+                return mMetrics.measuresLayoutDuration;
+            case  NUMBER_OF_VARIABLES:
+                return mMetrics.mVariables;
+            case  NUMBER_OF_EQUATIONS:
+                return mMetrics.mEquations;
+            case  NUMBER_OF_SIMPLE_EQUATIONS:
+                return mMetrics.mSimpleEquations;
+        }
+        return 0;
+    }
+
+    /** get a simple name for a statistic
+     *
+     * @param type type of statistic
+     * @return a camel case
+     */
+    String geName(int type) {
+        switch (type) {
+            case NUMBER_OF_LAYOUTS:
+                return "NumberOfLayouts";
+            case NUMBER_OF_ON_MEASURES:
+                return "MeasureCalls";
+            case NUMBER_OF_CHILD_VIEWS:
+                return "ChildCount";
+            case NUMBER_OF_CHILD_MEASURES:
+                return "ChildrenMeasures";
```

Comment:
nit: I am not sure if this is for the number of measures calls made by children. If so, maybe we could name it **ChildrenMeasuresCalls**, or  **NumberOfMeasures** to match line 216, or **WidgetsMeasuresCalls** to match DURATION_OF_CHILD_MEASURES.

Reply:
One is the 
A.  Number of  times ConstraintLayout called child.measure()
number of calls to child measure or number of child.measure()'s
The other is 
 B. The time spend in children doing measure.
or duration spent in children.measure() 

"Child Measures Calls" implies the children called measure. They might because they might have children but we do not measure that. 

Children have a method measure()  which ConstraintLayout called 
The Child spends time in Measure.

Anyway it was intentional

Reply:
Makes sense. Thanks for the explanation!